Functions/Admin/Backup-RsEncryptionKey.ps1

```diff
@@ -4,71 +4,105 @@
+            Use the "Connect-RsReportServer" function to set/update a default value.
+        
+        .PARAMETER Credential
+            The credentials with which to connect to the Report Server.
+            Use the "Connect-RsReportServer" function to set/update a default value.
+        
+        .EXAMPLE
+            Backup-RSEncryptionKey -Password 'Enter Your Password' -KeyPath 'C:\ReportingServices\Default.snk'
+            Description
+            -----------
+            This command will back up the encryption key against default instance from SQL Server 2016 Reporting Services
+        
+        .EXAMPLE
+            Backup-RSEncryptionKey -ReportServerInstance 'SQL2012' -ReportServerVersion '11' -Password 'Enter Your Password' -KeyPath 'C:\ReportingServices\Default.snk'
+            Description
+            -----------
+            This command will back up the encryption key against named instance (SQL2012) from SQL Server 2012 Reporting Services
+        
+        .EXAMPLE
+            Backup-RSEncryptionKey -ComputerName "sql2012a243" -Credential (Get-Credential) -ReportServerInstance 'SQL2012' -ReportServerVersion 'SQLServer2012' -Password 'Enter Your Password' -KeyPath 'C:\ReportingServices\Default.snk'
+            Description
+            -----------
+            This command will back up the encryption key against named instance (SQL2012) from SQL Server 2012 Reporting Services.
+            To do so, it will not use the default connection, but rather connect to the computer "sql2012a243", prompting the user for connection credentials.
     #>
 
-    [cmdletbinding()]
+    [CmdletBinding(SupportsShouldProcess = $true, ConfirmImpact = 'Medium')]
     param
     (
+        [Parameter(Mandatory = $True)]
         [string]
-        $SqlServerInstance='MSSQLSERVER',
-
+        $Password,
+        
+        [Parameter(Mandatory = $True)]
         [string]
-        $SqlServerVersion='13',
-
-        [Parameter(Mandatory=$True)]
+        $KeyPath,
+        
+        [Alias('SqlServerInstance')]
         [string]
-        $Password,
+        $ReportServerInstance,
+        
+        [Alias('SqlServerVersion')]
+        [Microsoft.ReportingServicesTools.SqlServerVersion]
+        $ReportServerVersion,
         
-        [Parameter(Mandatory=$True)]
         [string]
-        $KeyPath    
+        $ComputerName,
+        
+        [System.Management.Automation.PSCredential]
+        $Credential
     )
-
-    $rsWmiObject = New-RsConfigurationSettingObject -SqlServerInstance $SqlServerInstance -SqlServerVersion $SqlServerVersion
-
-    Write-Verbose "Retrieving encryption key..."
-    $encryptionKeyResult = $rsWmiObject.BackupEncryptionKey($Password)
-
-    if ($encryptionKeyResult.HRESULT -eq 0) 
-    {
-        Write-Verbose "Success!"
-    }
-    else 
+    
+    if ($PSCmdlet.ShouldProcess((Get-ShouldProcessTargetWmi -BoundParameters $PSBoundParameters), "Retrieve encryption key and create backup in $KeyPath"))
```

Comment:
Got it and fixed it.
Bad parameter type requirement (it's an internal type, which the parser couldn't detect), which I didn't catch because I tested things out inside my toolkit (where it's available), instead of a 100% regular console - gotta remind myself to try it in vanilla too, sorry for the inconvenience.